controllers/machinehealthcheck_controller.go

```diff
@@ -151,7 +176,55 @@ func (r *MachineHealthCheckReconciler) reconcile(_ context.Context, cluster *clu
 		UID:        cluster.UID,
 	})
 
-	return ctrl.Result{}, fmt.Errorf("controller not yet implemented")
+	logger := r.Log.WithValues("machinehealthcheck", m.Name, "namespace", m.Namespace)
+	logger = logger.WithValues("cluster", cluster.Name)
+
+	// Create client for target cluster
+	clusterClient, err := remote.NewClusterClient(ctx, r.Client, cluster, r.scheme)
+	if err != nil {
+		logger.Error(err, "Error building target cluster client")
+		return ctrl.Result{}, err
+	}
+
+	if err := r.watchClusterNodes(ctx, r.Client, cluster); err != nil {
+		logger.Error(err, "Error watching nodes on target cluster")
+		return ctrl.Result{}, err
+	}
+
+	// fetch all targets
+	logger.V(3).Info("Finding targets")
+	targets, err := r.getTargetsFromMHC(clusterClient, cluster, m)
+	if err != nil {
+		logger.Error(err, "Failed to fetch targets from MachineHealthCheck")
+		return ctrl.Result{}, err
+	}
+	totalTargets := len(targets)
+	m.Status.ExpectedMachines = int32(totalTargets)
+
+	// Default to 10 minutes but override if set in MachineHealthCheck
+	timeoutForMachineToHaveNode := defaultTimeoutForMachineToHaveNode
+	if m.Spec.NodeStartupTimeout != nil {
+		timeoutForMachineToHaveNode = m.Spec.NodeStartupTimeout.Duration
+	}
```

Comment:
Given that the defaulting webhook sets `m.Spec.NodeStartupTimeout`, can we use its value without redefaulting here?

Reply:
Yep, removed! I think I was trying to guard against, if for any reason the webhook defaulting hadn't worked, is that worth worrying about?